### DISCUSSION TITLE
Leistungslimit erscheint nicht in der Konsole der openDTU

### DISCUSSION THREAD

**Discussion by Alexander1714** [tbnobody/OpenDTU#2118] [Q&A]
Hallo an die MQTT openDTU-Speziallisten!

Ich komme mit meinem Problem einfach nicht weiter. Ich möchte die Leistung meiner Hoymiles-WR über den Homeassistant regeln. Die gewünschte Leistung wird allerdings nicht an die openDTU übertragen. Für die Fehlersuche habe ich den MQTT-Explorer verwendet.
Über den Explorer sende ich z.B.: solar/opendtuxxxxxx/SN vom WR/cmd/limit_nonpersistent_relative
und als payload gebe ich 33 als raw-Wert ein.

In der Konsole der openDTU fehlt aber der relative Wert 33%.

10:50:26.965 > Success
10:50:29.323 > Received MQTT message on topic: solar/opendtuxxxxxx/SN vom WR/cmd/limit_nonpersistent_relative
10:50:30.686 > Fetch inverter: xxxx

Dabei ist es egal ob der Wert als relative, absolute, persistent oder non-persistent gesendet wird.
Interessanterweise kann ich aber die WR problemlos mit dem Payload 0 oder 1 aus-, bzw. einschalten. Damit habe ich eine Regelung auch hinbekommen. Nicht schön, funktioniert aber.

Ich habe dazu schon alles mögliche durchsucht. Keiner scheint aber diese Problem zu haben. Ist mein Topic vielleicht zu lang? Wo liegt mein Fehler?

Vielen Dank!
Alexander

**Comment by Omega13x** [comment]
Müsste es nicht so aussehen?

solar/1141xxxxxxxx/cmd/limit_nonpersistent_relative

Zumindest steht es so in der Anleitung.

**Reply by Alexander1714** [reply]
Hallo Omega13x, im MQTT-Explorer steht unter Publish: solar/opendtuxxxxxx/1161833xxxx/cmd/limit_nonpersistent_relative
Den Payload habe ich mit z.B.: 10 als raw eingegeben (Format xml und json funktionieren eben so wenig).
Kommt genauso in der Konsole an, halt ohne das Payload.

**Comment by tbnobody** [comment]
Wenn als Base Topic `solar/opendtuxxxxxx` eingetragen wurde, dann ist die topic `solar/opendtuxxxxxx/SN vom WR/cmd/limit_nonpersistent_relative` richtig. Du musst aber auf jeden Fall aufpassen, dass keine Leerzeichen am Ende enthalten sind.

**Reply by Alexander1714** [reply]
Im  MQTT-Explorer kommt alles richtig ohne Leerzeichen an, z.B.: 
cmd
limit_nonpersistent_relative=33
Ich stehe echt auf dem Schlauch, vor allen weil das Kommando zum WR mit pwr 0 oder 1 funktioniert.

**Reply by Alexander1714** [reply]
Hallo tbnobody, habe eben auf die v24.6.29 aktualisiert. Hat aber keine Auswirkung auf mein Problem. Wäre es möglich das mein Topic zu viele Zeichen enthält und dann kein Platz mehr für das Payload vorhanden ist? Ich habe ja nach solar/ noch 25 Zeichen bis zur SN des WR eingefügt. Nicht das ich mir hier ein Eigentor schieße....

**Comment by Alexander1714** [comment]
Hallo an die Spezialisten, ich habe es auch über die Karte in Homeassistant getestet: Restart, Turn Off/ON funktionieren, Limits werden nicht übertragen. Damit scheiden Leerzeichen und Tippfehler im MQTT-Explorer aus. Die Entitäten kommen ja alle so von der openDTU.
Schon komisch, ich tappe da im Dunkeln. Hat keiner mehr eine Idee was ich noch probieren könnte?
![HM-WR-Karte in HA](https://github.com/tbnobody/OpenDTU/assets/139920930/e09b097c-bddf-478f-8b9b-7f6a6c79fb47)
